ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-interactions/location-interaction.tsx

```diff
@@ -14,16 +14,39 @@ const CqlUtils = require('../../../js/CQLUtils')
 const wreqr = require('wreqr')
 const Query = require('../../../js/model/Query')
 
+const createFilterTree = (locations: string[]) => {
+  const filterTree = {
+    type: 'OR',
+    filters: [] as any[],
+  }
+  filterTree.filters = locations.map((location: string) => {
+    return {
+      property: 'anyGeo',
+      type: 'INTERSECTS',
+      value: location,
+    }
+  })
+  return filterTree
+}
+
+const createCqlString = (locations: string[]) => {
+  const cqlArray = locations.map((location: string) => {
+    const geometry = Geometry.parse(location)
+    return CqlUtils.buildIntersectCQL(geometry)
+  })
+  return cqlArray.join(` OR `)
+}
+
 const handleCreateSearch = (props: Props) => {
   const locations = getGeoLocations(props.model)
 
   if (locations.length === 0) return
 
-  const locationString = `(${locations.join(` OR `)})`
   const newQuery = new Query.Model({
     type: locations.length > 1 ? 'advanced' : 'basic',
   })
-  newQuery.set('cql', locationString)
+  newQuery.set('cql', createCqlString(locations))
+  newQuery.set('filterTree', createFilterTree(locations))
```

Comment:
✏️ Any reason not to use `transformCQLToFilter` or `transformFilterToCQL` from `CQLUtils.js` to avoid manually creating either the filter tree or CQL string here?

Reply:
I originally had it like that, but that function wasn't actually creating the filters correctly when there was an `OR` in the CQL String. i.e. if there were multiple geometries on a single metacard